lib/testsuite/context.js

```diff
@@ -316,9 +316,13 @@ class Context extends EventEmitter {
       throw new Error(`The "${testName}" test script must be a function. "${typeof testFn}" given.`);
     }
 
-    // TODO: warn if test name already exists
     if (!skipTest) {
-      this.tests.push(testName);
+      if(this.tests.includes(testName)) {
+        throw new Error(`${testName} step already exists, testcases must have unique names in the test suite.`);
+      }
```

Comment:
This needs to be put outside of the `if` block because if we have a test suite with testcases like below:
```js
it('some title', function f1() {});
it.skip('some title', function f2() {});
```
while the second testcase will be skipped, the first testcase will run the callback of the second testcase (`f2`) instead of its own callback.